rusk-recovery/src/state/snapshot.rs

```diff
@@ -0,0 +1,61 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_bls12_381_sign::PublicKey;
+use serde_derive::{Deserialize, Serialize};
+
+mod acl;
+mod balance;
+mod stake;
+mod wrapper;
+
+use acl::Acl;
+pub use balance::Balance;
+pub use stake::GenesisStake;
+use wrapper::Wrapper;
+
+#[derive(Serialize, Deserialize, Default, PartialEq, Eq)]
+pub struct Snapshot {
+    acl: Acl,
+    balance: Vec<Balance>,
+    stake: Vec<GenesisStake>,
+}
+
+impl Snapshot {
+    pub fn add_balance(&mut self, balance: Balance) {
+        self.balance.push(balance);
+    }
+
+    pub fn add_stake(&mut self, stake: GenesisStake) {
+        self.stake.push(stake);
+    }
+
+    pub fn add_stake_owner<T: Into<PublicKey>>(&mut self, pk: T) {
+        let pk = pk.into();
+        self.acl.stake.owners.push(pk.into());
+    }
+
+    pub fn allow_staker<T: Into<PublicKey>>(&mut self, pk: T) {
+        let pk = pk.into();
+        self.acl.stake.allowlist.push(pk.into());
+    }
+
+    pub fn genesis_transfers(&self) -> impl Iterator<Item = &Balance> {
+        self.balance.iter()
+    }
+
+    pub fn active_stakes(&self) -> impl Iterator<Item = &GenesisStake> {
+        self.stake.iter()
+    }
+
+    pub fn stake_owners(&self) -> impl Iterator<Item = &PublicKey> {
+        self.acl.stake.owners.iter().map(|pk| &**pk)
+    }
+
+    pub fn allowlist(&self) -> impl Iterator<Item = &PublicKey> {
+        self.acl.stake.allowlist.iter().map(|pk| &**pk)
+    }
+}
```

Comment:
Missing docs everywhere.
Also, it looks like we have different objects that should have it's own object (e.g. stake). Let's discuss that in pairing.